op3_ball_detector/src/ball_detector.cpp

```diff
@@ -155,16 +167,17 @@ void BallDetector::process()
     // image filtering
     filterImage(img_hsv, img_filtered);
 
-    //std::cout << "TEST: " << img_filtered.empty() << std::endl;
-
     //detect circles
     houghDetection2(img_filtered);
   
     // was the detection valid?
     if (goodDetectionMode()) {
       std::cout << "Good detection mode for " << light_val << std::endl;
-      params_color_.adjustWeightsWithLightVal(light_val, 1, light_weights_);
+      params_color_.adjustWeightsWithLightVal(light_val, 4, light_weights_);
```

Comment:
I'm fine with it, and it might depend on how you crafted the model, by why increase by four?

Reply:
If the penalty and reward are the same, the distribution will never change unless it starts out really really good. Good color ranges are less common, and so are rewarded more. Bad ranges are more common, and are penalized less. Also, this means if the ball moves into a different light range, it won't take forever for the range to adjust to the new light level. However, these should be in constants. I will update them to be constants.